src/client/auth.js

```diff
@@ -22,6 +73,34 @@ module.exports = function (client, options) {
       }
     }
 
+    if (!options.session && options.profilesFolder) {
+      let auths = await getLP()
+
+      let lowerUsername = options.username.toLowerCase()
+      let profile = Object.keys(auths.authenticationDatabase).find(key => 
+        auths.authenticationDatabase[key].username.toLowerCase() == lowerUsername
+        || Object.values(auths.authenticationDatabase[key].profiles)[0].displayName.toLowerCase() == lowerUsername
+      )
+
+      if (profile) {
+        let newUsername = auths.authenticationDatabase[profile].username
+        let uuid = Object.keys(auths.authenticationDatabase[profile].profiles)[0]
+        let displayName = auths.authenticationDatabase[profile].profiles[uuid].displayName
+        let newProfile = {
+          name: displayName,
+          id: uuid
+        };
+
+        options.session = {
+          accessToken: auths.authenticationDatabase[profile].accessToken,
+          clientToken: auths.clientToken,
+          selectedProfile: newProfile,
+          availableProfiles: [newProfile]
+        }
+        options.username = newUsername
```

Comment:
Can we not set `client.session` and `client.username` instead like in the `cb` used in yggdrasilClient.validate ?

Reply:
No because we don't want code duplication and we'll load these efforts off to the rest of the code that also handles refreshing the token.

Reply:
Oh ok :smile: